### PR TITLE
bt-250: Create file upload endpoint

### LINKED ISSUE #250: Create file upload endpoint
We need to create endpoint to upload our files in the system connected with AWS S3 bucket
- [x] create account on AWS S3 bucket
- [x] create fileStorage package to upload file to AWS S3 (AWS keys to .env and update .env.example) 

  **BUT DON'T PUSH YOUR KEY TO GIT HUB!!!** 

- [x] create file repository 
- [x] create file service
- [x] create file controller

### PR DESCRIPTION
**Pull request by JosuerBague**
Closes #250 

Whats been done:

- file-storage using AWS S3
- file controller
- file service
- file repository
- Attached fastify-multer upload as prehandler to controller
- tested using CURL

Test Command with response: (Upload an image)
image

File Entities Saved to DB:
image

Files Available in S3:
image

Issues: I am not sure if my swagger annotation is correct, particularly the request body.

### CODE REVIEW

File: backend/src/migrations/20230906094421_update_files_table.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const TABLE_NAME = 'files';
+
+const ColumnName = {
+    ID: 'id',
+    URL: 'url',
+    NEW_FILE_NAME: 'file_name',
+    OLD_FILE_NAME: 'name',
+    CONTENT_TYPE: 'content_type',
+    CREATED_AT: 'created_at',
+    UPDATED_AT: 'updated_at',
```
Comment (MaksymGavatuk) [inline backend/src/migrations/20230906094421_update_files_table.ts]: If you don't use CONTENT_TYPE, CREATED_AT, etc remove these fields

File: backend/src/migrations/20230906094421_update_files_table.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    E_TAG: 'etag',
+};
```
Comment (MaksymGavatuk) [inline backend/src/migrations/20230906094421_update_files_table.ts]: In this case create OldColumnName = { ... } NewColumnName = { ... }

File: backend/.env.example [hunk 3]
```diff
@@ -23,3 +23,9 @@ DB_POOL_MAX=10
 # CRYPT
 #
 PASSWORD_SALT_ROUNDS=10
+
+#
+# AWS S3
+#
+AWS_ACCESS_KEY_ID=string
+AWS_SECRET_ACCESS_KEY=string
```
Comment (MaksymGavatuk) [inline backend/.env.example:36]: Forgot AWS_BUCKET_NAME

File: backend/src/common/packages/file-storage/file-storage.package.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    public constructor() {
+        this.s3 = new AWS.S3({
+            credentials: {
+                accessKeyId: config.ENV.AWS.AWS_ACCESS_KEY_ID,
+                secretAccessKey: config.ENV.AWS.AWS_SECRET_ACCESS_KEY,
+            },
+        });
+        this.bucketName = config.ENV.AWS.AWS_BUCKET_NAME;
+    }
```
Comment (MaksymGavatuk) [inline backend/src/common/packages/file-storage/file-storage.package.ts]: Pass this variables like constructor params

**Review by MaksymGavatuk — CHANGES_REQUESTED** [review]

**Comment by JosuerBague** [comment]
Retested image and file upload using curl, post-changes:

Pdf
image

Png:
image

Files are in bucket:
image

Entity details are stored in db:
image

**Review by MaksymGavatuk — APPROVED** [review]
Looks good for me

**Review by ptarho — APPROVED** [review]

**Comment by nikita-remeslov** [comment]
Conflicts

File: backend/.env.example
Comment (nikita-remeslov) [inline backend/.env.example:1]: EOL

File: backend/src/bundles/files/file.entity.ts
Comment (nikita-remeslov) [inline backend/src/bundles/files/file.entity.ts:1]: why do we need this changes?

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: backend/src/bundles/files/file.entity.ts
Comment (JosuerBague) [reply backend/src/bundles/files/file.entity.ts:1]: Hello, I discussed this in Slack with @MaksymGavatuk and in the end it was decided that for files only the following columns were needed:
- id
- url
- created_at
- updated_at
- etag
- file_name

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**